### DISCUSSION TITLE
New to HyperSnips, Installed HyperSnips, but can't get it to work. Help Appreciated 🙏

### DISCUSSION THREAD

**Discussion by davidleejy** [draivin/hsnips#190] [Q&A]
Hi, thanks for popping by this thread. I'm new to hypersnips but can't seem to get even the basic snippets to work; not observing any "hints of life" sparking from HyperSnips. Part of me wonders if I'm using HyperSnips correctly. Would be grateful if anyone could offer some help 🥲

What I've tried:

   - Adding flags `i` and `A` to the snippets.
   - Downgrading to Hypersnips versions 0.2.8 and 0.2.7.
   - Reloading snippets, reloading VSCode.
   - Pressing `tab` after typing trigger.
   - Typing spaces before trigger in an attempt to get HyperSnips to "kick in".

My VSCode version:
Version: 1.86.0 (Universal)
Commit: 05047486b6df5eb8d44b2ecd70ea3bdf775fd937
Date: 2024-01-31T10:29:15.765Z
Electron: 27.2.3
ElectronBuildId: 26495564
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.2.0

**Comment by FluffLescure** [comment]
1. Are you trying to implement the snippets for a latex project or for something else (C/C++, JS)?
If you're using it for C/C++ project then I can't help you since I'm having the same issue, and I believe it's related to Intellisense overwritting the regular suggested snippets.
If for latex check that the name for the snippet file is `latex.hsnips`.
For other environments like JS, rename the snippets file `all.hsnips`

2. Is the snippet file at the correct PATH ?
You can set it up in `settings.json` using `"hsnips.hsnipsPath" : "path/to/snippet/dir"`

**Reply by davidleejy** [reply]
1. I intend to apply snippets when editing .typ files (Typst language). Typst is a modern language that aims to do what latex does but isn't in the family of tex variants. Thanks for your suggestion about renaming the snippets file to `all.hsnips`. Will try this. Would you happen to have come across other ways of telling hypersnips to only apply snippets in an `all.hsnips` file to files belonging to a certain format?

2. I'm using the default path so the "hsnips.hsnipsPath" is empty.

**Reply by FluffLescure** [reply]
For context matching on different languages/files I recommend checking out : https://github.com/draivin/hsnips/discussions/139
